src/components/utils/PortalProvider.tsx

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import type {RefObject} from 'react';
+
+export const PortalContext = React.createContext<RefObject<HTMLElement>>({current: null});
```

Comment:
Let context value be just HTMLElement, with `document.body` default. And I would name it `PortalContainerContext`